CHANGES_3.0.0.md

```diff
@@ -78,3 +78,36 @@ Following [changes to the MongoDB connection string specification](https://githu
 For example, whereas before `mongodb://u$ername:pa$$w{}rd@/tmp/mongodb-27017.sock/test` would have been a valid connection string (with username `u$ername`, password `pa$$w{}rd`, host `/tmp/mongodb-27017.sock` and auth database `test`), the connection string for those details would now have to be provided to MongoClient as `mongodb://u%24ername:pa%24%24w%7B%7Drd@%2Ftmp%2Fmongodb-27017.sock/test`.
 
 For more information about connection strings, read the [connection string specification](https://github.com/mongodb/specifications/blob/master/source/connection-string/connection-string-spec.rst).
+
+
+## BulkWriteResult & BulkWriteError
+When errors occured with bulk write operations in the past, the driver would callback or reject with
+the first write error, as well as passing the resulting `BulkWriteResult`.  For example:
+
+```
+MongoClient.connect('mongodb://localhost', function(err, client) {
+  const collection = client.db('foo').collection('test-collection')
+
+  collection
+    .insert({ id: 1 })
+    .then(() => collection.insert({ id: 1 }))
+    .then(result => /* deal with errors in `result */);
+});
+```
+
+becomes:
+
+```
+MongoClient.connect('mongodb://localhost', function(err, client) {
+  const collection = client.db('foo').collection('test-collection')
+
+  collection
+    .insert({ id: 1 })
+    .then(() => collection.insert({ id: 1 }))
+    .catch(err => /* deal with errors in `err */);
```

Comment:
so if you want to do something with the BulkWriteResult in the case of no error, you'd just add another .then(..) to this chain? maybe worth adding that to the example?

Reply:
so the code would be:
```
  collection
    .insert({ id: 1 })
    .then(() => collection.insert({ id: 1 }))
    .then(result => /* deal with write result */)
    .catch(err => /* deal with errors in `err`, `err` also has the original result in `err.result`*/);
```